_posts/en/newsletters/2019-05-14-newsletter.md

```diff
@@ -0,0 +1,626 @@
+wiki page for changes -->{% endcomment %}
+
+[BIP173][] forbids bech32 addresses from using mixed cases.  The
+preferred way to write a bech32 address is in all lowercase, but there's
+one case where all uppercase makes sense: QR codes.  Take a look at the
+following two QR codes for the same address with the only difference
+being lowercase versus uppercase:
+
+    FIXME: image
+
+This was a deliberate design feature of bech32.  QR codes can be created
+in several modes (most notably: numeric, uppercase alphanumeric, and
+binary), each one using more bits to represent each character.  Legacy
+addresses require mixed case, so their only option is to use the binary
+mode.  Bech32 addresses for Bitcoin[^only-bc] can use the more efficient
+uppercase alphanumeric character set when the address is capitalized,
+allowing the QR code to be less complex.
+
+Simple [BIP21][] URIs can also use all uppercase characters with bech32
+because [RFC3986][] specifies the URI scheme name is case insensitive.
+
+    FIXME: image
+
+Unfortunately, the `?` and `&` needed for passing parameters in a BIP21
+URI are not part of the QR code uppercase character set, so our QR
+encoder automatically switches back to binary mode when they are used.
+
+    FIXME: image
+
+Still, anywhere you want to display just an address in a QR code, it
+makes sense to use all-uppercase for bech32 addresses and [bech32-like
+addresses][News 44 bech32].
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].*
+
+- [Bitcoin Core #15730][] extends the `getwalletinfo` RPC with a
+  `scanning` field that tells the user how far along the program
+  is in rescanning the block chain for transactions affecting their
+  wallet (if the user requested a rescan or it was automatically
+  triggered).  Otherwise, it simply returns `false`.
+
+- [Bitcoin Core #15930][] deprecates the `getunconfirmedbalance` RPC and
+  the three different *balance* fields in the `getwalletinfo` RPC.  In
+  their place, a `getbalances` RPC is added that provides provides two
+  sets of fields.  For balances the wallet can spend ("IsMine"), a
+  `trusted` field for outputs either created by the wallet itself or
+  which have at least one confirmation; an `untrusted_pending` field for
+  outputs in the mempool created by other users; and an `immature` field
+  for outputs from a miner generation transaction that haven't yet
+  received 100 confirmations (the earliest they can be spent).  For
+  balances the wallet is only watching ("watchonly"), the same three
+  fields are provided under a different object.
+
+- [C-Lightning #2524][] records details about failed attempts to forward
+  payments to the database and will display details in the
```

Comment:
Suggest removing "to the database". "forward payments to the database" could be confusing.